compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1962,7 +1962,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 t.rebind(def.all_fields().map(|f| f.ty(self.tcx(), substs)).collect())
             }
 
-            ty::Opaque(def_id, substs) => {
+            ty::Opaque(def_id, substs) | ty::TyAlias(def_id, substs) => {
```

Comment:
this may require recursion for `TyAlias`, since a type alias can refer to another type alias.

Reply:
Maybe confirmation ensures this already happens by recursing on sub-obligations. 

I think this code has to do with auto traits. We might need a test that checks auto traits work correctly with chains of type aliases.

Reply:
`bound_type_of` is calling `type_of` internally (as you can see [here](https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_middle/ty/util.rs.html#650-652)). So if I understand correctly, there is no need to recurse on the type alias, right?

Reply:
uh, actually I think this is fine. 

So this function returns the sub-types that must be checked for the type to be sized, but it's ok for us to just go down one "layer", because we'll just loop and call this again if the alias points to another alias.